src/Aspire.Hosting/ProjectResourceBuilderExtensions.cs

```diff
@@ -491,4 +490,48 @@ private static void SetAspNetCoreUrls(this IResourceBuilder<ProjectResource> bui
             }
         });
     }
+
+    private static void SetBothPortsEnvVariables(this IResourceBuilder<ProjectResource> builder)
+    {
+        builder.WithEnvironment(context =>
+        {
+            builder.SetOnePortsEnvVariable(context, "HTTP_PORTS", "http");
+            builder.SetOnePortsEnvVariable(context, "HTTPS_PORTS", "https");
+        });
+    }
+
+    private static void SetOnePortsEnvVariable(this IResourceBuilder<ProjectResource> builder, EnvironmentCallbackContext context, string portEnvVariable, string scheme)
+    {
+        if (context.EnvironmentVariables.ContainsKey(portEnvVariable))
+        {
+            // If the user has already set that variable, we don't want to override it.
+            return;
+        }
+
+        var ports = new ReferenceExpressionBuilder();
+        var firstPort = true;
+
+        // Turn endpoint ports into a single environment variable
+        foreach (var e in builder.Resource.GetEndpoints().Where(e => IsValidAspNetCoreUrl(e.EndpointAnnotation)))
+        {
+            var port = e.EndpointAnnotation.TargetPort ?? e.EndpointAnnotation.Port;
```

Comment:
delete this.

Reply:
So we only care about TargetPort here, and don't want to fall back to Port, right?

Reply:
That's right.

Reply:
Done.